File: Parrot.cpp
#include <iostream>
#include <vector>
#include <random>

class Parrot {
    std::vector<std::string> phrases_;
public:
    Parrot(const std::string& phrase) {
        phrases_.emplace_back(phrase);
    }

    void say(int count = 1) const {
        for (int i = 0; i < count; ++i) {
            const std::string& phrase = phrases_[rand() % phrases_.size()];
            std::cout << phrase << " ";
        }
        std::cout << std::endl;
    }

    void setPhrase(const std::string& phrase) {
        phrases_.emplace_back(phrase);
    }
};

int main() {
    Parrot parrot("Hello");
    parrot.say();
    parrot.setPhrase("AAAA");
    parrot.setPhrase("BBBB");
    for (int i = 1; i <= 10; ++i)
        parrot.say(i);
}

File: Parrot.java
import java.util.Random;

public class Parrot {
    String[] phrases_ = new String[10];
    int count_, cap_ = 10;
    Random rand = new Random();

    public Parrot(String phrase) {
        phrases_[0] = phrase;
        count_ = 1;
    }

    public void say(int count) {
        for (int i = 0; i < count; i += 1) {
            int chosen = rand.nextInt(count_);
            System.out.println(phrases_[chosen]);
        }
    }

    public void setPhrase(String phrase) {
        if (count_ == cap_) {
            cap_ += 10;
            phrases_ = new String[cap_];
        }
        phrases_[count_] = phrase;
        count_ += 1;
    }

    public static void main(String[] args) {
        Parrot parrot = new Parrot("Hello");
        parrot.say(1);
        parrot.setPhrase("AAAA");
        parrot.setPhrase("BBBB");
        for (int i = 1; i <= 5; i++) {
            parrot.say(i);
        }
    } 
}